consensus/safety-rules/src/safety_rules.rs

```diff
@@ -462,6 +465,45 @@ impl SafetyRules {
         let signature = self.sign(timeout)?;
         Ok(signature)
     }
+
+    fn guarded_sign_commit_vote(
+        &mut self,
+        ledger_info: LedgerInfoWithSignatures,
+        new_ledger_info: LedgerInfo,
+    ) -> Result<Ed25519Signature, Error> {
+        self.signer()?;
+
+        // Verify that ledger_info contains at least 2f + 1 dostinct signatures
+        ledger_info
+            .verify_signatures(&self.epoch_state()?.verifier)
+            .map_err(|error| Error::InvalidQuorumCertificate(error.to_string()))?;
+
+        let old_ledger_info = ledger_info.ledger_info();
+
+        if old_ledger_info.commit_info().clone() == BlockInfo::empty() {
```

Comment:
don't need the clone, should be able to use *commit_info

Reply:
resolved